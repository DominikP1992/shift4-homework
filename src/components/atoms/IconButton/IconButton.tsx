import theme from '@/styles/theme';
import styled from '@xstyled/emotion';

export interface IconButtonProps
  extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  size?: 'small' | 'medium';
}

const IconButton = styled.button`
  border-radius: 5px;
  width: 24px;
  height: 24px;
  line-height: 20px;
  display: flex;
  justify-content: center;
  align-items: center;

  transition:
    background-color 0.2s,
    color 0.2s;

  background-color: transparent;

  &:hover {
    background-color: ${theme.colors['icon-button--hover']};
  }

  &:active {
    background-color: ${theme.colors['icon-button--pressed']};
  }

  &:focus {
    background-color: ${theme.colors['icon-button--pressed']};
    outline: none;
  }

  &:disabled {
    background-color: transparent;
    opacity: 0.5;
  }

  ${({ size }: Pick<IconButtonProps, 'size'>) =>
    size === 'medium' &&
    `
        width: 40px;
        height: 40px;
        border-radius: 10px;
    `}
`;

export default IconButton;
