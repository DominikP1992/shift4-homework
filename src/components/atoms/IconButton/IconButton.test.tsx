import React from 'react';
import { render, screen } from '@/tests/testsUtils';
import IconButton from './IconButton'; // Adjust the import path as necessary

describe('IconButton', () => {
  it('renders without crashing', () => {
    render(<IconButton aria-label="test button" />);
    expect(screen.getByLabelText('test button')).toBeInTheDocument();
  });

  it('match snapshot', () => {
    render(<IconButton aria-label="test button" />);
    expect(screen.getByLabelText('test button')).toMatchSnapshot();
  });

  it('applies medium size styles correctly', () => {
    const { getByRole } = render(<IconButton size="medium" />);
    expect(getByRole('button')).toHaveStyle('width: 40px');
    expect(getByRole('button')).toHaveStyle('height: 40px');
  });
});
