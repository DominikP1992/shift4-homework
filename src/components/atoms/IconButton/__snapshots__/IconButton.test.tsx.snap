// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton match snapshot 1`] = `
<button
  aria-label="test button"
  class="css-29n6zw"
/>
`;
